### PR TITLE
Free form text survey 

### PR DESCRIPTION
**Pull request by carocao-msft**
# What
Added some UI changes and added free form text survey

# Why
UI Changes: 
1. increase star sizes ( feedback from Alex P) 
2. move star ratings descriptions to below stars ( feedback from Alex P) 
3. Change categorys order to 'Overall, audio, video, presenting' ( feedback from Alex M) 
4. User are allowed to skip tag survey and submit feedback ( feedback from Alex M)
5. Show tag survey when star rating is <= 2 stars  ( feedback from Alex M)
6. Decrease panel width to 24 rem ( feedback from Alex P) 

# How Tested
Tested with samples 

# Process & policy checklist

- [ ] I have updated the project documentation to reflect my changes if necessary.
- [ ] I have read the [CONTRIBUTING](https://github.com/Azure/communication-ui-library/blob/main/CONTRIBUTING.md) documentation.

**Is this a breaking change?**

- [ ] This change causes current functionality to break.

### CODE REVIEW

File: packages/react-composites/src/composites/common/SurveyPane.tsx [hunk 1]
```diff
@@ -9,30 +9,134 @@ import { useLocale } from '../localization';
...
 export const SurveyPane = (props: {
   /* @conditional-compile-remove(end-of-call-survey) */
   onSubmitSurvey?: (survey: CallSurvey) => Promise<CallSurveyResponse | undefined>;
+  /* @conditional-compile-remove(end-of-call-survey) */
+  onSurveySubmittedCustom?: (
+    callId: string,
+    surveyId: string,
+    /**
+     * This is the survey results containing star survey data and API tag survey data.
+     * This part of the result will always be send to calling sdk
```
Comment (edwardlee-msft) [inline packages/react-composites/src/composites/common/SurveyPane.tsx:28]:
```suggestion
     * This part of the result will always be sent to the calling sdk
```

File: packages/react-composites/src/composites/localization/locales/en-US/strings.json [hunk 2]
```diff
@@ -241,6 +241,7 @@
     "starRatingAriaLabel": "Select {0} of {1} stars",
     "surveyCancelButtonAriaLabel": "Cancel",
     "tagsSurveyQuestion": "What could have been better?",
+    "tagsSurveyTextFieldDefaultText": "Other, please specify",
     "tagsSurveyHelperText": "Check any issues you experienced",
     "surveyTextboxDefaultText": "Other,please specify",
```
Comment (edwardlee-msft) [inline packages/react-composites/src/composites/localization/locales/en-US/strings.json:246]:
```suggestion
    "surveyTextboxDefaultText": "Other, please specify",
```

**Review by edwardlee-msft — APPROVED** [review]

File: packages/react-composites/src/composites/CallComposite/adapter/AzureCommunicationCallAdapter.ts [hunk 3]
```diff
@@ -741,7 +741,7 @@ export class AzureCommunicationCallAdapter<AgentType extends CallAgent | BetaTea
       this.callClient,
       this.callAgent,
       this.deviceManager,
-      undefined,
+      this.call,
```
Comment (JamesBurnside) [inline packages/react-composites/src/composites/CallComposite/adapter/AzureCommunicationCallAdapter.ts:744]: do we need to call `createHandlers` in a leave call scenario? Can we just leave them as is and delete this function call?

File: packages/react-composites/src/composites/common/SurveyPane.tsx [hunk 4]
```diff
@@ -9,30 +9,134 @@ import { useLocale } from '../localization';
...
 import { SurveyPaneContent } from './SurveyPaneContent';
+/* @conditional-compile-remove(end-of-call-survey) */
+import { CallSurveyImprovementSuggestions } from '@internal/react-components';
 
 /** @private */
 export const SurveyPane = (props: {
   /* @conditional-compile-remove(end-of-call-survey) */
   onSubmitSurvey?: (survey: CallSurvey) => Promise<CallSurveyResponse | undefined>;
+  /* @conditional-compile-remove(end-of-call-survey) */
+  onSurveySubmittedCustom?: (
```
Comment (JamesBurnside) [inline packages/react-composites/src/composites/common/SurveyPane.tsx:23]: what is the difference between onSubmitSurvey and onSurveySubmittedCustom?

Comment (JamesBurnside) [reply packages/react-composites/src/composites/common/SurveyPane.tsx:23]: Can they be rolled into one prop?

**Review by JamesBurnside — APPROVED** [review]

File: packages/react-composites/src/composites/common/SurveyPane.tsx (hunk 4 above)
Comment (carocao-msft) [reply packages/react-composites/src/composites/common/SurveyPane.tsx:23]: onSubmitSurvey comes from calling directly and onSurveySubmittedCustom is the function contoso pass in to get access to survey data. They take different props so can't be rolled into one

File: packages/react-composites/src/composites/CallComposite/adapter/AzureCommunicationCallAdapter.ts (hunk 3 above)
Comment (carocao-msft) [reply packages/react-composites/src/composites/CallComposite/adapter/AzureCommunicationCallAdapter.ts:744]: Let me check who wrote this line of code and verify

Comment (carocao-msft) [reply packages/react-composites/src/composites/CallComposite/adapter/AzureCommunicationCallAdapter.ts:744]: Prabhjot wrote this in this PR https://github.com/Azure/communication-ui-library/pull/3352/files we should probably keep this for now

**Pull request merged**